src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c

```diff
@@ -30,17 +31,93 @@ const ActorInit Bg_Ingate_InitVars = {
+
+    if ((globalCtx->sceneNum != 0x63 || gSaveContext.linkAge != 0) ||
+        (((gSaveContext.eventChkInf[1] & 0x100) != 0) && (gSaveContext.cutsceneIndex != 0xFFF0))) {
+        Actor_Kill(&this->dyna.actor);
+        return;
+    }
+
+    Actor_SetScale(&this->dyna.actor, 0.1f);
+    if (((this->dyna.actor.params & 1) != 0) && ((gSaveContext.eventInf[0] & 0xF) == 6)) {
+        globalCtx->csCtx.frames = (u16)0U;
+        BgIngate_SetupAction(this, &func_80892890);
+        return;
+    }
+    BgIngate_SetupAction(this, &func_80892990);
+}
+
+void BgIngate_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgIngate* this = THIS;
+
+    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, (s32)this->dyna.dynaPolyId);
+}
+
+void func_80892890(BgIngate* this, GlobalContext* globalCtx) {
+    s32 phi0;
+    s16 phi1;
+    s16 ctxFrames;
+
+    if (globalCtx->csCtx.frames >= 0x32) {
+        phi0 = 0x4000;
+        if ((this->dyna.actor.params & 2) == 0) {
+            phi0 = -0x4000;
+        }
+        this->dyna.actor.shape.rot.y = this->dyna.actor.posRot.rot.y + phi0;
+        BgIngate_SetupAction(this, &func_80892990);
+    } else if (globalCtx->csCtx.frames >= 0xA) {
+        ctxFrames = globalCtx->csCtx.frames - 0xA;
+        ctxFrames *= 400;
+        phi1 = ctxFrames;
+        if (ctxFrames > 0x4000) {
+            ctxFrames = 0x4000;
+        }
+        ctxFrames = (Math_Sins(ctxFrames) * 16384.0f);
+        phi1 = ctxFrames;
+        if ((this->dyna.actor.params & 2) == 0) {
+            phi1 = -phi1;
+        }
+        this->dyna.actor.shape.rot.y = this->dyna.actor.posRot.rot.y + phi1;
+    }
+}
+
+void func_80892990(BgIngate* this, GlobalContext* globalCtx) {
+    // does nothing
+}
+
+void BgIngate_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgIngate* this = THIS;
+    this->actionFunc(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ingate/BgIngate_Destroy.s")
+void BgIngate_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
+    Gfx* dispRefs[4];
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ingate/func_80892890.s")
+    Graph_OpenDisps(dispRefs, globalCtx->state.gfxCtx, "../z_bg_ingate.c", 0xF0);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ingate/func_80892990.s")
+    func_80093D18(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ingate/BgIngate_Update.s")
+    gSPMatrix(gfxCtx->polyOpa.p++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_bg_ingate.c", 0xF5),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ingate/BgIngate_Draw.s")
+    gSPDisplayList(gfxCtx->polyOpa.p++, D_06001040);
+    Graph_CloseDisps(dispRefs, globalCtx->state.gfxCtx, "../z_bg_ingate.c", 0xFA);
```

Comment:
the hex numbers at the end of both of these lines are actually line numbers, so they should be in decimal

Reply:
Done... could you tell me why it's making reference to a source file and a specific line? Is this debug info or something?

Reply:
yes, its debug info for the debug rom. they likely used __LINE__ or whatever the directive is that gets the line automatically